### ISSUE TITLE
Suggested Retail Price (SRP, MSRP, RRP, etc) core or attribute

### ISSUE THREAD

**Issue by objecttothis** [opensourcepos/opensourcepos#2472]
### New Feature

Most manufacturers include a Recommended Retail Price, Suggested Retail Price, Manufacturer's Suggested Retail Price or similar with their products.  This information can be helpful for retailers with pricing and helpful for customers to see discount over SRP.  The information can be stored currently as an attribute, but I'm thinking it would be helpful to have SRP as an optional field in the items table because the information could then be used in reports to make suggestions on pricing changes to retailers and be used in the receipt printing to show customers how much they saved over the SRP prices of their items.

Thoughts?

I'm willing to do the work to get SRP into the database since I would also be sending it in my CLCdesq API integration.  Then anyone can expand on it or they can wait for me to get around to implementing the receipt printing features and price suggestion reporting.

**Comment by WebShells** [comment]
I think showing the cashier what was the price of this item for this client last sale would help in discount process also... Am not sure if SRP will be helpful for client or cashier more...

**Comment by SteveIreland** [comment]
Would #551 also be good to include with this change or has that been resolved with custom attributes?

**Comment by objecttothis** [comment]
@WebShells it probably won't help the cashier, it definitely helps the client to see amount saved. On the business side it probably helps most with pricing adjustments. For example, a report could show items with retail price over SRP to make it easy for buyers to reduce markup to meet or go below SRP. A different report could show items significantly under SRP which helps buyers to increase markup if desired.

**Comment by objecttothis** [comment]
@SteveIreland a limitation of our current Attributes implementation is that it lends itself to storing data but not performing operations on that data because the attribute may not exist with every business.

This is the reason why this issue, #551 and issues like them would need to be part of the core items table.  #551 could use the SRP as part of it's rule to have a default markup but could be implemented without this issue the way that the system is currently setup.

The short answer is that I don't think #551 is obsolete or resolved by Attributes.

**Comment by daN4cat** [comment]
Thinking in more abstract terms, aren't all of these discount policies in the end?

**Comment by objecttothis** [comment]
@daN4cat the SRP by itself isn't a discount policy, but can be used in pricing decisions, reporting and receipting.  #551 does fit more into discount policy or discount matrix structure.  The SRP is more of a building block of information.  If I was inclined to just keep it as information and not actually do something with it such as show it on receipts or create reports showing Items above or significantly below SRP then I would probably just say make it an attribute.  The problem is that our attribute system doesn't really allow for operations to be performed on an attribute since the attribute may not exist in an installation.

**Comment by daN4cat** [comment]
Ok, got it. It's more a double price policy. Probably a price in foreign currency can be close to this.

**Comment by daN4cat** [comment]
A generic solution would be to have multiple sales price and currencies and make it selectable with default the standard price.

**Comment by daN4cat** [comment]
@SteveIreland implemented something related to foreign currencies. Not sure that can be expanded? Never had the time to test it.

**Comment by objecttothis** [comment]
@daN4cat I think we might be mixing two many changes into this issue.  Having the value of the SRP stored in items has multiple enhancements attached to it, but by itself has nothing to do with multiple currencies, discount matrices or markup calculations.  I think this issue should just include getting the SRP into items as a non-required field and displaying on the receipt the amount saved over SRP (optionally) if the SRP exists for an item.

Then in separate enhancements if people want to use the SRP in their calculations for discounts, markup, etc then its already available.  I just think it's too complicated to put all the enhancements possible into one pull request to add SRP into items.

**Comment by daN4cat** [comment]
Right, I thought to have seen a ship in the fog but in reality it was an iceberg.... bang ;-)
